clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java

```diff
@@ -248,29 +249,34 @@ private RecordsIterator(LogEntry entry) {
 
             long wrapperRecordTimestamp = entry.record().timestamp();
             this.logEntries = new ArrayDeque<>();
+            // If relative offset is used, we need to decompress the entire message first to compute
+            // the absolute offset. For simplicity and because it's a format that is on its way out, we
+            // do the same for message format version 0
             try {
                 while (true) {
                     try {
                         LogEntry logEntry = getNextEntryFromStream();
-                        Record recordWithTimestamp = new Record(
-                            logEntry.record().buffer(),
-                            wrapperRecordTimestamp,
-                            entry.record().timestampType()
-                        );
-                        logEntries.add(new LogEntry(logEntry.offset(), recordWithTimestamp));
+                        if (entry.record().magic() > Record.MAGIC_VALUE_V0) {
```

Comment:
No, the check is to decide if the message format is 0 or 1. Message format 1 includes a timestamp and relative offsets and was introduced in Kafka 0.10.0.0.